web_pwa_oca/controllers/main.py

```diff
@@ -1,75 +1,156 @@
+        )
+        background_color = config_param_sudo.get_param(
+            "pwa.manifest.background_color", "#2E69B5"
+        )
+        theme_color = config_param_sudo.get_param("pwa.manifest.theme_color", "#2E69B5")
+        return {
+            "name": pwa_name,
+            "short_name": pwa_short_name,
+            "icons": self._get_pwa_manifest_icons(pwa_icon),
+            "start_url": '/web',
+            "display": "standalone",
+            "background_color": background_color,
+            "theme_color": theme_color,
+        }
 
-    @route('/web_pwa_oca/manifest.json', type='http', auth="public")
-    def manifest(self):
-        qweb = request.env['ir.qweb'].sudo()
-        config_param = request.env['ir.config_parameter'].sudo()
-        pwa_name = config_param.get_param("pwa.manifest.name", "Odoo PWA")
-        pwa_short_name = config_param.get_param("pwa.manifest.short_name", "Odoo PWA")
-        icon128x128 = config_param.get_param(
-            "pwa.manifest.icon128x128",
-            "/web_pwa_oca/static/img/icons/icon-128x128.png")
-        icon144x144 = config_param.get_param(
-            "pwa.manifest.icon144x144",
-            "/web_pwa_oca/static/img/icons/icon-144x144.png")
-        icon152x152 = config_param.get_param(
-            "pwa.manifest.icon152x152",
-            "/web_pwa_oca/static/img/icons/icon-152x152.png")
-        icon192x192 = config_param.get_param(
-            "pwa.manifest.icon192x192",
-            "/web_pwa_oca/static/img/icons/icon-192x192.png")
-        icon256x256 = config_param.get_param(
-            "pwa.manifest.icon256x256",
-            "/web_pwa_oca/static/img/icons/icon-256x256.png")
-        icon512x512 = config_param.get_param(
-            "pwa.manifest.icon512x512",
-            "/web_pwa_oca/static/img/icons/icon-512x512.png")
-        background_color = config_param.get_param(
-            "pwa.manifest.background_color", "#2E69B5")
-        theme_color = config_param.get_param(
-            "pwa.manifest.theme_color", "#2E69B5")
-        mimetype = 'application/json;charset=utf-8'
-        content = qweb.render('web_pwa_oca.manifest', {
-            'pwa_name': pwa_name,
-            'pwa_short_name': pwa_short_name,
-            'icon128x128': icon128x128,
-            'icon144x144': icon144x144,
-            'icon152x152': icon152x152,
-            'icon192x192': icon192x192,
-            'icon256x256': icon256x256,
-            'icon512x512': icon512x512,
-            'background_color': background_color,
-            'theme_color': theme_color,
-        })
-        return request.make_response(content, [('Content-Type', mimetype)])
+    @route("/web_pwa_oca/manifest.webmanifest", type="http", auth="public")
+    def pwa_manifest(self):
+        """Returns the manifest used to install the page as app"""
+        return request.make_response(
+            json.dumps(self._get_pwa_manifest()),
+            headers=[("Content-Type", "text/javascript;charset=utf-8")],
```

Comment:
Shouldn't it be something like this?

```suggestion
            headers=[("Content-Type", "text/json;charset=utf-8")],
```